### PR TITLE
Bump teslajsonpy to fix #275

### LINKED ISSUE #275: API breakage in version 2022.36 (it does not seem to include charge_to_max_range anymore)
## Version of the custom_component

2.4.3

## Describe the bug

API breakage in version 2022.36, it does not seem to include charge_to_max_range anymore

This causes crash that also causes other entries becoming unavailable

I have created a patch for this on teslajsonpy that fixes this issue, see https://github.com/zabuldon/teslajsonpy/pull/351

## Debug log

```text

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 151, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 283, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 110, in async_update_listeners
    update_callback()
  File "/config/custom_components/tesla_custom/tesla_device.py", line 129, in refresh
    self.tesla_device.refresh()
  File "/usr/local/lib/python3.10/site-packages/teslajsonpy/homeassistant/charger.py", line 121, in refresh
    self.__maxrange_state = data["charge_to_max_range"]
KeyError: 'charge_to_max_range'

```

### PR DESCRIPTION
**Pull request by mariogrip**

### CODE REVIEW

**Comment by alandtse** [comment]
Please also update pyproject.toml and run 'poetry update'. We use poetry to track dependencies.

**Comment by mariogrip** [comment]
> Please also update pyproject.toml and run 'poetry update'. We use poetry to track dependencies.

Sure, will do

**Comment by mariogrip** [comment]
Done @alandtse

**Comment by alandtse** [comment]
You need to update it in pyproject.toml first then run poetry update to pick it up.

File: poetry.lock [hunk 1]
```diff
@@ -1470,15 +1477,18 @@ pbr = ">=2.0.0,<2.1.0 || >2.1.0"
...
+python-versions = ">=3.7"
+
+[package.extras]
+tests = ["pytest-cov", "pytest"]
 
 [[package]]
 name = "teslajsonpy"
-version = "2.4.3"
+version = "2.4.5"
 description = "A library to work with Tesla API."
```
Comment (mariogrip) [inline poetry.lock:1492]: It did pick it up here, no?

**Comment by mariogrip** [comment]
> You need to update it in pyproject.toml first then run poetry update to pick it up.

It was set as a not hard version dependency, it is set to version higher then 2.0.0? It did pick it up in the lock file. But i can change it to a hard version requirement if you want, but maybe that should be a seperate pr?

**Comment by alandtse** [comment]
Yes, please update the hard dependency. It can be this PR.

**Comment by mariogrip** [comment]
> Yes, please update the hard dependency. It can be this PR.

Done 👍

**Pull request merged**